src/main/java/org/folio/innreach/controller/ContributionCriteriaController.java

```diff
@@ -0,0 +1,69 @@
+package org.folio.innreach.controller;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+import org.folio.innreach.domain.service.ContributionCriteriaConfigurationService;
+import org.folio.innreach.dto.ContributionCriteriaDTO;
+
+import org.folio.innreach.mapper.ContributionCriteriaMapperImpl;
+import org.folio.innreach.rest.resource.ContributionCriteriaApi;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.UUID;
+
+@Log4j2
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/inn-reach/central-servers/")
+public class ContributionCriteriaController implements ContributionCriteriaApi {
+
+  private final ContributionCriteriaConfigurationService criteriaConfigurationService;
+  private final ContributionCriteriaMapperImpl contributionCriteriaMapper;
+
+  @Override
+  @PostMapping("/contribution-criteria")
+  public ResponseEntity<ContributionCriteriaDTO> postContributionCriteria(
+    @Valid @RequestBody ContributionCriteriaDTO contributionCriteriaDTO) {
+
+    return ResponseEntity.status(HttpStatus.CREATED).body(contributionCriteriaMapper.toContributionCriteriaDTO(criteriaConfigurationService
```

Comment:
Usually if a service accepts and returns DTO objects then it also makes the necessary conversions inside. Thus I'd recommend to move conversion completely inside the service

Reply:
Especially added for making "weak" link between "Controller" and  "Service" DTO conversion layer has been moved to "Service" layer.